### ISSUE TITLE
[android] undefined is not an object (evaluating I.default.ERRORS)

### ISSUE THREAD

**Issue by nicolazj** [expo/expo#4474]
## 🐛 Bug Report
I guess the error are from 

`export const { ERRORS, SCOPES, TYPES } = ExpoGoogleSignIn; ` 
 
### Environment
ExpoKit, Android apk file, using SDK 33.  tested in genymotion android 9 .

### Steps to Reproduce

`import * as Google from 'expo-google-sign-in';`

![screenshot-2019-06-07_14 17 04 871](https://user-images.githubusercontent.com/1446380/59077689-bbe00600-892f-11e9-9a9c-4301f462685c.png)

**Comment by ankhzet** [comment]
Same with `expo-localization`. Correctly runs on device in dev mode, but built .apk reports mentioned error.

**Comment by ide** [comment]
@nicolazj @ankhzet -- it looks like the Google sign-in and location modules aren't installed in your standalone APK. This will happen if you have not added the modules you use to the `dependencies` in your package.json file.

1. Firstly, the original message in this issue says the import looks like `import * as Google from 'expo-google-sign-in'`. This is good -- importing from `expo` is now deprecated for most APIs.
2. `expo-google-sign-in` must be listed under package.json's dependencies. If you used the expo-codemod script to rewrite your imports, it printed out a command to run: `expo install expo-google-sign-in`. Run this command to install the package in your project; `expo install` will select the correct version of the package.
3. After you have updated package.json, run `expo build:android` again. The builders look at your dependencies to determine which of the Expo modules to include.

**Comment by brentvatne** [comment]
to people who encounter this error: make sure that for every `expo-*` import you have that you have also installed the appropriate package.

for example, in sdk32 i had:

```
import { WebBrowser } from 'expo';
```

now in sdk33 i changed it to:

```
import * as WebBrowser from 'expo-web-browser';
```

then i ran:

```
expo install expo-web-browser
```

**Comment by brentvatne** [comment]
we have now updated our android build service to behave the same as before the sdk33 release, so it includes all of the expo sdk rather than just the parts that have been installed in package.json.

including only the parts of the sdk that you use will be made default again at some point in the future after more testing and more work on providing better warnings. you can opt back into it now using `expo build:android -t app-bundle`, but 1) this only works on sdk32 or 33 and 2) you need to enable app signing in the google play developer console to do so.

**Comment by SimenB** [comment]
Thanks for reverting 🙂 

As an extra data point - the codemod did not detect `FontAwesome.loadFont()` from our `import { FontAwesome } from '@expo/vector-icons';` and install whatever I need for that to work. I've tried adding `expo-font` and `expo-asset` to `package.json` now, is that enough? Since the changes were reverted, I'll just build a new apk for now

**Comment by alamothe** [comment]
I had to `expo install` `react-native-maps` and `react-native-svg` after upgrading. Is this expected?

Also, `expo install` adds these lines to `package.json`:

```
    "react-native-maps": "^0.24.2",
    "react-native-svg": "^9.4.0",
```

I'm wondering if version should be made exact instead?

**Comment by WahidHasanAriyanto** [comment]
guys, how do you capture the error message on standalone app? errors happend in my application are never been shown and the app is just restarted itself
everything goes fine when i ran my projects in development (expo client)

**Comment by stevematdavies** [comment]
I have the same issue, and the same question as Wahid above,

**Comment by ankhzet** [comment]
@WahidHasanAriyanto, @stevematdavies As of SDK 35 iirc, fatal errors are not displayed in the generic error screen, app simply crashes now.

Refer to https://github.com/expo/expo/issues/4230 for more info.

Now it's mandatory for a dev to provide custom error boundary component into your app if you want to handle uncaught react lifecycle errors and show the user some custom error ui, and all dangling promises and callbacks must have their own `catch` handlers. 
Simply put, if some code throws an exception and that exception isn't handled explicitly by the developer, it would be considered a fatal exception and would crash the app.

Here's the relative link on React's Error Boundaries:
https://reactjs.org/docs/error-boundaries.html
